api/dbadapters/bigquery.ts

```diff
@@ -257,54 +248,82 @@ export class BigQueryDbAdapter implements IDbAdapter {
     return persistedMetadata;
   }
 
-  public async persistStateMetadata(actions: dataform.IExecutionAction[]): Promise<void> {
-    if (actions.length === 0) {
+  public async persistStateMetadata(
+    allActions: dataform.IExecutionAction[],
+    actionsToPersist: dataform.IExecutionAction[],
+    options: {
+      onCancel: OnCancel;
+    }
+  ): Promise<void> {
+    if (allActions.length === 0) {
       return;
     }
-    const allInvolvedTargets = new StringifiedSet(JSONObjectStringifier.create<dataform.ITarget>());
-    actions.forEach(action => {
-      allInvolvedTargets.add(action.target);
-      action.transitiveInputs.forEach(transitiveInput => allInvolvedTargets.add(transitiveInput));
-    });
-
-    const tableMetadataByTarget = new StringifiedMap<dataform.ITarget, dataform.ITableMetadata>(
-      JSONObjectStringifier.create()
-    );
-    await Promise.all(
-      Array.from(allInvolvedTargets).map(async target => {
-        tableMetadataByTarget.set(target, await this.table(target));
-      })
-    );
-    const queries = actions.map(action => {
-      const persistTable = dataform.PersistedTableMetadata.create({
-        target: action.target,
-        lastUpdatedMillis: tableMetadataByTarget.get(action.target).lastUpdatedMillis,
-        definitionHash: hashExecutionAction(action),
-        transitiveInputTables: action.transitiveInputs.map(transitiveInput => {
-          if (!tableMetadataByTarget.has(transitiveInput)) {
-            throw new Error(`Could not find table metadata for ${JSON.stringify(transitiveInput)}`);
-          }
-          return tableMetadataByTarget.get(transitiveInput);
-        })
-      });
-
-      const targetName = `${action.target.database}.${action.target.schema}.${action.target.name}`;
-
-      return buildQuery(targetName, persistTable);
-    });
+    const targetStringifier = JSONObjectStringifier.create<dataform.ITarget>();
```

Comment:
sure we don't want to just write a custom stringifier for Target? this will write longer than necessary and hard to read columns, is it worth just writing something custom that is similar to getActionName?

Reply:
honestly, my fear is that any custom stringifier will miss out any new fields we add in the future. but sure, done